src/ViewController.swift

```diff
@@ -0,0 +1,188 @@
+        helpPopover.behavior = .semitransient
+
+        // For better UI
+        view.window!.standardWindowButton(.miniaturizeButton)!.isHidden = true
+        view.window!.standardWindowButton(.zoomButton)!.isHidden = true
+        view.window!.styleMask.insert(.resizable)
+
+        DispatchQueue.main.async {
+            self.arrayController.content = self.resolutions[0]
+        }
+    }
+
+    override func viewDidAppear() {
+        super.viewDidAppear()
+        view.window?.level = .floating // Always on top
+    }
+
+    @IBAction func add(_ sender: Any) {
+        resolutions[0].append(Resolution())
+        arrayController.content = resolutions[0]
+        arrayController.rearrangeObjects()
+    }
+
+//    @IBOutlet weak var removeButton: NSButton!
+
+    @IBAction func remove(_ sender: Any) {
+        if arrayController.selectionIndexes.count > 0 {
+            let removed = arrayController.selectionIndexes.map { resolutions[0][$0] }
+
+            // For backward compatibility
+            if resolutions.count > 1 {
+                for rem in removed {
+                    if let j = resolutions[1].firstIndex(where: { $0 == rem }) {
+                        resolutions[1].remove(at: j)
+                    }
+                }
+            }
+
+            resolutions[0].remove(at: arrayController.selectionIndexes)
+            arrayController.content = resolutions[0]
+            arrayController.rearrangeObjects()
+        }
+    }
+
+    @IBAction func save(_ sender: Any) {
+        var saveScripts = [String]()
+
+        if ViewController.rootWriteable {
+            if let error = RestoreSettingsItem.backupSettings(originalPlistPath: fileNames.last!) {
+                NSAlert(fromDict: error).beginSheetModal(for: view.window!)
+                return
+            }
+        }
+
+        for ((plist, resol), (dir, fileName)) in zip(zip(plists, resolutions), zip(dirs, fileNames)) {
+            let tmpFile = NSTemporaryDirectory() + UUID().uuidString
+
+            plist.setValue(displayProductName as NSString, forKey: kDisplayProductName)
+            plist.setValue(resol.map { $0.toData() } as NSArray, forKey: "scale-resolutions")
+            plist.write(toFile: tmpFile, atomically: false)
+
+            saveScripts.append("mkdir -p '\(dir)'")
+            saveScripts.append("cp '\(tmpFile)' '\(fileName)'")
+            saveScripts.append("rm '\(tmpFile)'")
+        }
+
+        if let error = NSAppleScript.executeAndReturnError(source: saveScripts.joined(separator: " && "),
+                                                           asType: .shell,
+                                                           withAdminPriv: true) {
+            NSAlert(fromDict: error).beginSheetModal(for: view.window!)
+        } else {
+            view.window!.close()
+        }
+    }
+
+    @IBAction func calcAspectRatio(_ sender: Any) {
+        presentAsSheet(calcController)
```

Comment:
The sheet should have a Cancel button. Currently it's possible to close it without adding a new resolution only using Esc key.

Reply:
I'll add one soon. Thanks for feedback!

Reply:
Resolved in jnooree/RDM@e99669b1727cc5fe6508a1655f46f4a6c78bb050.